os/fs/smartfs/smartfs_smart.c

```diff
@@ -807,10 +811,16 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 				goto errout_with_semaphore;
 			}
 
-			/* Now get the chained sector info and reset the offset */
-
-			sf->curroffset = sizeof(struct smartfs_chain_header_s);
-			sf->currsector = SMARTFS_NEXTSECTOR(header);
+			/* If file modified in this loop but buf if buf data is remain, it will try to append data.
+			 * In this case, next sector of last sector is empty. So do not set header data Here, 
+			 * It will handled when it sync
+			 */
+			if (SMARTFS_NEXTSECTOR(header) != 0xFFFF) {
+				sf->currsector = SMARTFS_NEXTSECTOR(header);
+				
+				/* Now get the chained sector info and reset the offset */
+				sf->curroffset = sizeof(struct smartfs_chain_header_s);
+			}
```

Comment:
Do we need a `break;` statement here?
In case `SMARTFS_NEXTSECTOR(header) != 0xFFFF` or otherwise?

Reply:
If nextsector is 0xffff it means that it reaches lastsector.
Because loop checks sf->filepos < sf->entry.datlen, datlen calculated refer to used of chainheader of each sector when file opened.
So nextsector is 0xffff means that filepos reaches lastsector, it will break by condition of loop =)